packages/aws-cdk-lib/aws-events/lib/event-bus.ts

```diff
@@ -325,9 +336,14 @@ export class EventBus extends EventBusBase {
 
     super(scope, id, { physicalName: eventBusName });
 
+    if (props?.description && !Token.isUnresolved(props.description) && props.description.length > 512) {
```

Comment:
I'm curious why we're checking for a token in this field. If we're letting users pass in the description as a token, will this catch whether that token resolves to a value > 512 characters?

I think we should also include a unit test or two making sure tokenized values are resolved properly, and that tokenized strings > 512 characters throw the same expected error as a non-tokenized string.

Reply:
@Leo10Gama 
Thank you for your suggestion.

I believe we should not add the unit test you mentioned for the following two reasons:

- It is unlikely that the `description` will be passed as a token. Of course, this does not mean that such cases do not exist.

- Tokenized values are resolved during the CDK lifecycle, particularly during the deployment stage, not during the synthesis stage. Their validation is already handled by CFn and the respective AWS services, and I don't believe it's the responsibility of CDK unit tests. In fact, the CDK user guide (https://docs.aws.amazon.com/cdk/v2/guide/tokens.html#tokens_encoding) states that "If name is a token, validation isn't performed, and an error could still occur in a later stage in the lifecycle, such as during deployment," indicating that errors are expected during the deployment lifecycle when a token is passed.

@go-to-k, do you have any thoughts?

Reply:
This process says that if the property is a token, no validation will be performed.
In other words, this validation is skipped in the CDK layer, so I think unit testing is not necessary.

Although it is unlikely that the token itself will exceed 512 characters, the developer may mimic this validation process when other properties are newly added. Considering such a case, I think it is worthwhile to include such a process (skip tokens during validation).

Reply:
That makes sense, thanks for the clarification!